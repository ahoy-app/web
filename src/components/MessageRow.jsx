import React from 'react'
import PropTypes from 'prop-types'

import Message from './Message'

const styles = {
  padding: '5px 0px 5px 0px',
  boxSizing: 'border-box',

  // alignSelf: 'flex-end',
}

const MessageRow = ({ message }) => {
  const me = message.from === '-'
  return (
    <div
      style={{
        ...styles,
        alignSelf: me ? 'flex-end' : 'flex-start',
      }}
    >
      <Message message={message} me={me} />
    </div>
  )
}

MessageRow.propTypes = {
  message: PropTypes.object.isRequired,
}

export default MessageRow
