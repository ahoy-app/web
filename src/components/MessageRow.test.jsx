import React from 'react'
import { shallow } from 'enzyme'

import MessageRow from './MessageRow'

const message = {
  id: 'q384tp1349fgh139p48g',
  from: 'Ted',
  content: 'See you later',
}

describe('MessageRow', () => {
  it('should render correctly', () => {
    const component = shallow(<MessageRow message={message} />)
    expect(component).toMatchSnapshot()
  })
})
