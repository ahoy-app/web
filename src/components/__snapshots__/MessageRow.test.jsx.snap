// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageRow should render correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "alignSelf": "flex-start",
      "boxSizing": "border-box",
      "padding": "5px 0px 5px 0px",
    }
  }
>
  <Message
    message={
      Object {
        "content": "See you later",
        "from": "Ted",
        "id": "q384tp1349fgh139p48g",
      }
    }
  />
</div>
`;
